### PR TITLE
Sql db seeding 2

### PR DESCRIPTION
**Pull request by tyler-thee-creator**
PR for CRUD API refactoring along with testing, also refactored way data is sent to client and confirmed front end is not broken

Some files may have small changes, but you can just focus on these:

- database/data_generator.js
- database/query_generator.js
- database/database.js
- server/index.js
- test/api_tests.test.js

### CODE REVIEW

File: database/data_seeding/query_generator.js [hunk 1]
```diff
@@ -28,4 +24,77 @@ module.exports.addForeignKey = (table, fk, referenceTable, referenceKey) => {
...
+    cg.id_encid AS cg_id_encid,
+    cg.title AS cg_title,
+    cg.description AS cg_description
+
+    FROM features AS f
+    INNER JOIN features_list AS fl
+    ON f.product_id = fl.product_id
+    INNER JOIN content_grid_feature_items AS cg
+    ON f.product_id = cg.product_id
+
```
Comment (samasno) [inline database/data_seeding/query_generator.js:75]: these aliases definitely make the query a lot more readable

File: database/data_seeding/query_generator.js [hunk 2]
```diff
@@ -28,4 +24,77 @@ module.exports.addForeignKey = (table, fk, referenceTable, referenceKey) => {
...
+  }
+  const updates = keyValues.join(', ');
+
+  const query = `UPDATE ${table} SET ${updates} WHERE product_id = ${productId} RETURNING *;`;
+
+  console.log(query);
+
+  return query;
+}
+
```
Comment (samasno) [inline database/data_seeding/query_generator.js:95]: I haven't used that RETURNING key word, ill have to add that into my queries

File: database/data_seeding/query_generator.js [hunk 3]
```diff
@@ -28,4 +24,77 @@ module.exports.addForeignKey = (table, fk, referenceTable, referenceKey) => {
...
+    cg.description AS cg_description
+
+    FROM features AS f
+    INNER JOIN features_list AS fl
+    ON f.product_id = fl.product_id
+    INNER JOIN content_grid_feature_items AS cg
+    ON f.product_id = cg.product_id
+
+    WHERE f.product_id = ${productId};`;
+
```
Comment (sam-masno) [inline database/data_seeding/query_generator.js:77]: aliases really keep this query clean

File: database/database.js [hunk 4]
```diff
@@ -1,94 +1,72 @@
...
-      title: String,
-      description: String
-    }]
+  try {
+    await client.connect();
+    const res = await client.query(query);
+    await client.end();
+    return res;
+  } catch (err) {
+    return err;
```
Comment (samasno) [inline database/database.js:28]: is it necessary/best practice to create a new connection on every query?

File: database/data_seeding/query_generator.js [hunk 5]
```diff
@@ -28,4 +24,77 @@ module.exports.addForeignKey = (table, fk, referenceTable, referenceKey) => {
...
+  console.log(query);
+
+  return query;
+}
+
+module.exports.generateDeleteFeatureQuery = (table, productId) => {
+  const query = `DELETE FROM ${table} WHERE product_id = ${productId} RETURNING product_id;`;
+
+  return query;
```
Comment (samasno) [inline database/data_seeding/query_generator.js:99]: good clean query generators

**Review by samasno — APPROVED** [review]
left a few comments, looks good

**Pull request merged**